### DISCUSSION TITLE
Cut down font to my design

### DISCUSSION THREAD

**Discussion by Johnwf01john** [olikraus/u8g2#1735] [Q&A] [answered]
I am using 2 fonts in arduino micro, run out of space. I need to cut down fonts to what I need. the fonts I am using are.
u8g2_font_ncenR10_tr - I only need numbers and X, Y
u8g2_font_5x7_tr - Only need A B C D E F M N O S T  a b c d e h i m n p s t /

I need to know if there is documentation to explain how to achieve this and how to install the new fonts into code. I realise there is some information here, but not in one place or clear.  I have Fony, never used it yet. and I believe I need BDFconv but do not know how to use this or how to instal it on a PC.

**Comment by olikraus** [comment]
Did you also check the u8g2 FAQ?

**Reply by olikraus** [reply]
https://github.com/olikraus/u8g2/blob/61c71b28b116129e01d8750fd90d4d94fbe1ca0d/doc/faq.txt#L245-L273

**Reply by olikraus** [reply]
Let me provide a step by step example for the ncenR10 font (windows assumed). 
1. Download bdfconv.exe from https://github.com/olikraus/u8g2/tree/master/tools/font/bdfconv
2. Download ncenR10 bdf from https://github.com/olikraus/u8g2/blob/master/tools/font/bdf/ncenR10.bdf (RAW Button)
3. Start cmd.exe and within the terminal window execute bdfconv.exe: `bdfconv  -v -f 1 -m "48-57,88,89" ncenR10.bdf -n u8g2_font_ncenR10 -o test.c`
4. The font will be stored in test.c, but you can just paste the content of test.c in your .ino file.

The arguments for -m are numbers (48-57), X (88) and Y (89).

![ncenR10](https://user-images.githubusercontent.com/1084992/147251070-f847f5f0-3550-40d8-a535-b2c606d37f16.png)

The resulting font occupies only 226 bytes, compared to 1585 bytes of the "tr" version:

![https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_ncenR10_tr.png](https://github.com/olikraus/u8g2/wiki/fntpic/u8g2_font_ncenR10_tr.png)

**Reply by Johnwf01john** [reply]
Thanks this has worked great, where do you get u8g2_font_5x7_tr.bdf? and I may be able to cut it down.

Thanks for a very speedy response. :)

**Comment by olikraus** [comment, marked as answer]
The corresponding BDF file is here: https://github.com/olikraus/u8g2/blob/master/tools/font/bdf/5x7.bdf
... if this had been your question.

**Reply by Johnwf01john** [reply]
Thanks that was my question, where can I get BDF fonts or fonts to change into BDF?

Many thanks for your help and fast response.

**Reply by Johnwf01john** [reply]
I hope you have a merry christmas!

**Reply by olikraus** [reply]
> Thanks that was my question, where can I get BDF fonts or fonts to change into BDF?
> 

BDF fonts, used for u8g2, are all listed here: https://github.com/olikraus/u8g2/tree/master/tools/font/bdf
TTF can be converted to BDF with OTF2BDF tool (http://sofia.nmsu.edu/~mleisher/Software/otf2bdf/) 
Fony.exe can export to BDF. Also fontforge can export to BDF.

> I hope you have a merry christmas!

I miss some snow here in Germany.
